core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -815,6 +815,39 @@ public static int bitLength(ByteString s) {
     return s.length() * 8;
   }
 
+  /** SQL BIT_GET(value, position) function. */
+  public static int bitGet(long value, int position) {
```

Comment:
Shall we take other number types like Decimal into consideration?

Reply:
According to Spark's [PR](https://github.com/apache/spark/pull/31198/files) to implement BIT_GET function, currently we don't support decimal type. For consistency, I think we should not take other number types like Decimal into consideration.